libnavigation-core/src/test/java/com/mapbox/navigation/core/infra/FakeNativeNavigator.kt

```diff
@@ -0,0 +1,176 @@
+package com.mapbox.navigation.core.infra
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.bindgen.Expected
+import com.mapbox.common.TileStore
+import com.mapbox.common.TilesetDescriptor
+import com.mapbox.navigation.base.options.DeviceProfile
+import com.mapbox.navigation.base.options.PredictiveCacheLocationOptions
+import com.mapbox.navigation.navigator.internal.MapboxNativeNavigator
+import com.mapbox.navigation.navigator.internal.NativeNavigatorRecreationObserver
+import com.mapbox.navigator.BannerInstruction
+import com.mapbox.navigator.CacheHandle
+import com.mapbox.navigator.ElectronicHorizonObserver
+import com.mapbox.navigator.Experimental
+import com.mapbox.navigator.FallbackVersionsObserver
+import com.mapbox.navigator.FixLocation
+import com.mapbox.navigator.GraphAccessor
+import com.mapbox.navigator.HistoryRecorderHandle
+import com.mapbox.navigator.NavigationStatus
+import com.mapbox.navigator.NavigationStatusOrigin
+import com.mapbox.navigator.NavigatorConfig
+import com.mapbox.navigator.NavigatorObserver
+import com.mapbox.navigator.PredictiveCacheController
+import com.mapbox.navigator.RoadObjectMatcher
+import com.mapbox.navigator.RoadObjectsStore
+import com.mapbox.navigator.RoadObjectsStoreObserver
+import com.mapbox.navigator.RouteAlternativesControllerInterface
+import com.mapbox.navigator.RouteInfo
+import com.mapbox.navigator.RouterError
+import com.mapbox.navigator.RouterInterface
+import com.mapbox.navigator.TilesConfig
+import io.mockk.mockk
+
+/***
+ * Emulates the behavior of the native navigator for testing.
+ *
+ * [FakeNativeNavigator] doesn't cover all scenarios.
+ * If you find an empty method instead of behavior emulation feel free to implement it.
+ * Don't hesitate to write tests for this test double.
+ *
+ * Some behavior can't be easily emulated.
+ * It's hard to parse a route and emulate location to trigger the status updates.
+ * Use [FakeNativeNavigator] as a stub for status update emulation.
+ * Call [updateStatus] method to trigger all observers.
+ */
+class FakeNativeNavigator private constructor() : MapboxNativeNavigator {
```

Comment:
```suggestion
class TestNativeNavigator private constructor() : MapboxNativeNavigator {
```

Typically these classes are "Test*" classes. Referencing [TestCoroutineDispatcher](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-test-coroutine-dispatcher/index.html] [TestLifecycleOwner](https://developer.android.com/reference/kotlin/androidx/lifecycle/testing/TestLifecycleOwner),